test/index.test.js

```diff
@@ -0,0 +1,455 @@
+
+sinon.assert.expose(assert, { prefix: '' });
+
+const TEST_XML =
+`<project>
+<description/>
+<keepDependencies>false</keepDependencies>
+<properties/>
+<scm class="hudson.scm.NullSCM"/>
+<canRoam>true</canRoam>
+<disabled>false</disabled>
+<blockBuildWhenDownstreamBuilding>false</blockBuildWhenDownstreamBuilding>
+<blockBuildWhenUpstreamBuilding>false</blockBuildWhenUpstreamBuilding>
+<triggers/>
+<concurrentBuild>false</concurrentBuild>
+<builders>
+<hudson.tasks.Shell>
+<command>sleep 100 | echo 'Hello, I am going to sleep for 100s!'</command>
+</hudson.tasks.Shell>
+</builders>
+<publishers/>
+<buildWrappers/>
+</project>`;
+
+describe('index', () => {
+    let executor;
+    let Executor;
+    let fsMock;
+    let jenkinsMock;
+    let requestMock;
+    let getCrumbMock;
+    let jenkinsClientMock;
+    let initJenkinsClientMock;
+
+    const crumbUrl = 'http://admin:fakepassword@jenkins:8080/crumbIssuer/api/json';
+
+    const config = {
+        buildId: '4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e',
+        jobId: '50dc14f719cdc2c9cb1fb0e49dd2acc4cf6189a0',
+        pipelineId: 'ccc49349d3cffbd12ea9e3d41521480b4aa5de5f',
+        container: 'node:4',
+        scmUrl: 'git@github.com:screwdriver-cd/data-model.git#master'
+    };
+
+    const buildIdConfig = {
+        buildId: '4b8d9b530d2e5e297b4f470d5b0a6e1310d29c5e'
+    };
+
+    const fakeCrumb = {
+        statusCode: 200,
+        body: {
+            _class: 'hudson.security.csrf.DefaultCrumbIssuer',
+            crumb: '24e80888069a1beaa5af3e0e3ef201d0',
+            crumbRequestField: 'Jenkins-Crumb'
+        }
+    };
+
+    const fakeJobInfo = {
+        lastBuild: {
+            number: 1
+        }
+    };
+
+    before(() => {
+        mockery.enable({
+            useCleanCache: true,
+            warnOnUnregistered: false
+        });
+    });
+
+    beforeEach(() => {
+        fsMock = {
+            readFileSync: sinon.stub()
+        };
+
+        jenkinsMock = sinon.stub();
+
+        requestMock = sinon.stub();
+
+        jenkinsClientMock = {
+            job: {
+                create: sinon.stub(),
+                get: sinon.stub(),
+                build: sinon.stub()
+            },
+            build: {
+                stop: sinon.stub(),
+                log: sinon.stub()
+            }
+        };
+
+        fsMock.readFileSync.returns(TEST_XML);
+
+        mockery.registerMock('fs', fsMock);
+        mockery.registerMock('jenkins', jenkinsMock);
+        mockery.registerMock('request', requestMock);
+
+        /* eslint-disable global-require */
+        Executor = require('../index');
+
+        executor = new Executor({
+            host: 'jenkins',
+            username: 'admin',
+            password: 'fakepassword'
+        });
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+        mockery.resetCache();
+    });
+
+    after(() => {
+        mockery.disable();
+    });
+
+    it('extends base class', () => {
```

Comment:
I think this case is redundant. The `start`, `stream`, `stop` describe blocks infer that these are functions in order to assert the cases they're testing against.